src/libraries/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -240,6 +241,25 @@ internal static void Memmove(ref byte dest, ref byte src, nuint len)
             return;
 
         MCPY05:
+            // Misaligned access via SIMD is especially expensive on ARM64 on large data.
+            // 512 is an arbitrary threshold picked for Ampere and Apple M1.
+            //
+            // TODO: Consider doing the same on x86/AMD64 for V256 and V512
+#if HAS_CUSTOM_BLOCKS
+            if (Vector128.IsHardwareAccelerated && len >= 512)
+            {
+                // Try to opportunistically align the destination below. The input isn't pinned, so the GC
+                // is free to move the references. We're therefore assuming that reads may still be unaligned.
+                //
+                // dest is more important to align than src because an unaligned store is more expensive
+                // than an unaligned load.
+                nuint misalignedElements = Vector128.Size - (nuint)Unsafe.AsPointer(ref dest) & (Vector128.Size - 1);
+                Unsafe.As<byte, Block16>(ref dest) = Unsafe.As<byte, Block16>(ref src);
```

Comment:
```suggestion
                Unsafe.WriteUnaligned<Block16>(ref dest, Unsafe.ReadUnaligned<Block16>(ref src));
```

Reply:
The current impl is using `Unsafe.As` in the method everywhere so I didn't change it, presumably if an unaligned read is a problem then `HAS_CUSTOM_BLOCKS` should not be defined for such a platform (e.g. it currently is not defined for RISC-V and ARM32)